avocado/core/test.py

```diff
@@ -569,11 +569,7 @@ def _run_avocado(self):
             raise exceptions.TestSkipError(details)
         except exceptions.TestCancel as details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
-            skip_illegal_msg = ('Calling cancel() in setUp() '
-                                'is not allowed in avocado, you '
-                                'must fix your test. Original cancel exception: '
-                                '%s' % details)
-            raise exceptions.TestError(skip_illegal_msg)
+            raise
```

Comment:
Also why do you treat `TestCancel` differently than plain exception? I'd just remove this block and it'd get treated by `except` including `avocado.test` traceback, which is useful when looking for where it got actually skipped (there can be many locations with the same message)